src/bin/text/dune

```diff
@@ -1,9 +1,18 @@
 (documentation
   (package alt-ergo))
 
+(executable
+  (name gen_link_flags)
+  (libraries unix fmt stdcompat cmdliner)
+  (modules gen_link_flags)
+  (promote (until-clean)))
+
 (rule
- (with-stdout-to link_flags.dune
-  (run ./gen-link-flags.sh %{env:LINK_MODE=dynamic} %{ocaml-config:system})))
+ (target link_flags.dune)
+ (deps (:gen gen_link_flags.exe))
```

Comment:
I believe this is not needed and you can just use `./gen_link_flags.exe` (or `%{bin:gen_link_flags}`) in the `run` command below

Reply:
Right, `%{bin}` scope is for installed executable — the following should work with no promotion tho:

```
(executable
  (name gen_link_flags)
  (libraries unix fmt stdcompat)
  (modules gen_link_flags))

(rule
 (with-stdout-to link_flags.dune
  (run ./gen_link_flags.exe %{env:LINK_MODE=dynamic} %{ocaml-config:system})))
```